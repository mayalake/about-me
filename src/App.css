.App {
  font-family: palatino;
}

.Header{
	text-align: center;
	color: white;
	background: linear-gradient(rgba(0,0,0,.3), rgba(0,0,0,.3)), url(./img/background.jpg);
	background-position: center;
	background-repeat: no-repeat;
	background-size: cover;
	height: 100vh;
}

.HeaderText{
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
}

.Footer{
	text-align: center;
	color: grey;
	font-size: 12px;
}
