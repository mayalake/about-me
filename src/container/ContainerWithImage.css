.ContentContainer{
	width:100%;
}

.ImageContainer{
	display: inline-block;
	vertical-align: top;
	width: 20%;
	margin: 0 5% 0 0;
}
.isCircleImage{
	border-radius: 50%;
}

.ContentNextToImage{
	display: inline-block;
	width: 70%;
}