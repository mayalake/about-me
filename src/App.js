import React from 'react';

import NavBar from './navbar/NavBar';
import AboutMeSection from './about-section/AboutMeSection.jsx';
import ResumeSection from './resume-section/ResumeSection';
import ContactSection from './contact-section/ContactSection';

import './App.css';

const App = () =>
    <div className="App">
    	<header className="Header">
			<NavBar />
			<div className="HeaderText">
				<h1>Maya Lake</h1>
				<p>Software Developer<br/>Los Angeles</p>
			</div>
		</header>
		<AboutMeSection />
		<ResumeSection />
		<ContactSection />
		<footer className="Footer">&#169; 2020 Maya Lake</footer>
    </div>;

export default App;
