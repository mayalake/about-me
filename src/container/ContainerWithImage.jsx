import React from 'react';
import PropTypes from 'prop-types';

import SimpleContainer from './SimpleContainer';

import './ContainerWithImage.css';

const ContainerWithImage = (props) =>{
	return(
		<SimpleContainer id={props.id} name={props.name} isGreyBackground={props.isGreyBackground}>
			<div className="ContentContainer">
				<img className={`ImageContainer ${props.isCircleImage && "isCircleImage"}`} src={props.image} alt=""/>
				<div className="ContentNextToImage">
					{props.children}
				</div>
			</div>
		</SimpleContainer>
	);
}

ContainerWithImage.propTypes = {
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	image: PropTypes.string.isRequired,
	isGreyBackground: PropTypes.bool,
	isCircleImage: PropTypes.bool
}

export default ContainerWithImage;