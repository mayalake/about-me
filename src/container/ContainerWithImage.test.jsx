import React from 'react';
import { render } from '@testing-library/react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../App';
import ContainerWithImage from './ContainerWithImage';

Enzyme.configure({ adapter: new Adapter() });

const sectionDetail = {id:"id", name:"name", image:"image"};
let wrapped = Enzyme.shallow(<ContainerWithImage id={sectionDetail.id} name={sectionDetail.name} image={sectionDetail.image}/>);

describe('ContainerWithImage', () => {
	it('renders the ContainerWithImage correctly', () =>{
		expect(wrapped).toMatchSnapshot();
	});
});