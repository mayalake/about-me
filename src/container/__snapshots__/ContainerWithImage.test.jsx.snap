// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerWithImage renders the ContainerWithImage correctly 1`] = `ShallowWrapper {}`;
